rust/chains/tw_ton/src/signing_request/builder.rs

```diff
@@ -47,27 +46,54 @@ impl SigningRequestBuilder {
         })
     }
 
-    /// Currently, V4R2 wallet supported only.
-    fn wallet(input: &Proto::SigningInput) -> SigningResult<TonWallet<WalletV4>> {
+    /// Currently, V4R2 and V5R1 wallets supported.
+    fn wallet(input: &Proto::SigningInput) -> SigningResult<VersionedTonWallet> {
+        let wallet_version = input.wallet_version;
+        if wallet_version != Proto::WalletVersion::WALLET_V4_R2
+            && wallet_version != Proto::WalletVersion::WALLET_V5_R1
+        {
+            return SigningError::err(SigningErrorType::Error_not_supported)
+                .context("'WALLET_V4_R2' and 'WALLET_V5_R1' are supported only");
+        }
```

Comment:
There is no need for this check as the version will be matched below

Reply:
Thanks for pointing this out, I have removed this unnecessary check.